server/sonar-web/src/main/js/components/SourceViewer/components/LineCode.tsx

```diff
@@ -117,41 +112,44 @@ export default class LineCode extends React.PureComponent {
   attachEvents() {
     if (this.codeNode) {
       this.symbols = this.codeNode.querySelectorAll('.sym');
-      for (const symbol of this.symbols) {
-        symbol.addEventListener('click', this.handleSymbolClick);
+      if (this.symbols) {
+        for (let i = 0; i < this.symbols.length; i++) {
+          const symbol = this.symbols[i];
+          symbol.addEventListener('click', this.handleSymbolClick);
+        }
       }
     }
   }
 
   detachEvents() {
     if (this.symbols) {
-      for (const symbol of this.symbols) {
-        symbol.removeEventListener('click', this.handleSymbolClick);
+      for (let i = 0; i < this.symbols.length; i++) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Expected a 'for-of' loop instead of a 'for' loop with this simple iteration ([typescript:S4138](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4138))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fpopups&issues=AWHg9DkC3WeWj6zJbMvC&open=AWHg9DkC3WeWj6zJbMvC)

[ISSUE_KEY]: AWHg9DkC3WeWj6zJbMvC